packages/firebase/index.d.ts

```diff
@@ -7551,7 +7614,7 @@ declare namespace firebase.firestore {
      * to the backend (Note that it won't resolve while you're offline).
      */
     update(
-      documentRef: DocumentReference,
+      documentRef: DocumentReference<unknown>,
```

Comment:
Using `unknown` here breaks consumers of our SDK that are still on TypeScript 2.x. Can you confirm with the JS team that that is ok?

Reply:
Confirmed w/ Feiyang that they currently only support 2.8 and up.

Reply:
I think that means we have to drop "unknown".

Reply:
Oops didn't mean to resolve. Is the right thing to do to replace with `<any> ` and add a comment to disable linter?

Reply:
FWIW @schmidt-sebastian recently "discovered" that

```
function foo(x: any): Bar { ... }
```
is essentially equivalent to:
```
function foo<T>(x: T): Bar { ... }
```
TypeScript will let you pass any type as `x`...  and so maybe the right thing to do here is just make update generic as well:
```
update<T>(documentRef: DocumentReference<T>, ...)
```

Reply:
Looks like `any` is already used in the file without disabling tslint. I can't think of a better way to express this.

Reply:
It works in the `index.d.ts` files, but when I used it in `database.ts`, it would trigger the "unexpected any" error. Should I just disable tslint, or switch to the other notation? I think using `<any>` is more readable and easier to understand for the end user, but I'm not sure if there are any consequences to disabling tslint.

Reply:
Using `any` and disabling the tslint check (on a single-line basis) in database.ts should be fine.